include/swift/AST/Stmt.h

```diff
@@ -554,81 +552,102 @@ class alignas(1 << PatternAlignInBits) StmtConditionElement {
     CK_HasSymbol,
   };
 
+private:
+  llvm::PointerIntPair<void *, 2> OpaqueAndKind;
```

Comment:
Personally I would prefer to keep the `PointerUnion`, to make it explicit the underlying types we store, and avoid having to `static_cast` to the right type.